src/twisted/protocols/tls.py

```diff
@@ -682,6 +690,100 @@ def clientConnectionForTLS(self, protocol):
         return self._connectionForTLS(protocol)
 
 
+class AggregateSmallWrites:  # TODO make this private for now?
+    """
+    Aggregate small writes so they get written in large batches.
+
+    If this is used as part of a transport, the transport needs to call
+    ``flush()`` immediately when ``loseConnection()`` is called, otherwise any
+    buffered writes will never get written.
+    """
+
+    def __init__(self, write: Callable[[bytes], object], clock: IReactorTime):
+        self._write = write
+        self._clock = clock
+        self._buffer: list[bytes] = []
+        self._bufferLen = 0
+        self._scheduled: Optional[IDelayedCall] = None
+
+    def write(self, data: bytes) -> None:
+        """
+        Buffer the data, or write it immediately if we've accumulated enough to
+        make it worth it.
+
+        Accumulating too much data can result in higher memory usage.
+        """
+        self._buffer.append(data)
+        self._bufferLen += len(data)
+        if self._bufferLen > 64_000:
+            self.flush()
+        if self._scheduled is None:
+            self._scheduled = self._clock.callLater(0, self._scheduledFlush)
```

Comment:
Just minor comment that I hope will make the code easier to read

```suggestion
        if self._scheduled:
            # We already have a scheduled send,
            # so with the data in the buffer,
            # there is nothing more to do here.
            return

        # Schedule the write of the accumulated buffer
        # for the next reactor iteration.
        self._scheduled = self._clock.callLater(0, self._scheduledFlush)
```